### PR TITLE
WD-13742 - ci: set up CI for UI

### PR DESCRIPTION
**Pull request by huwshimi**
## Changes

Set up CI for UI files.

## Details

https://warthogs.atlassian.net/browse/WD-13742

### CODE REVIEW

File: tsconfig.json
Comment (huwshimi) [inline tsconfig.json:1]: This top level config allows IDEs to pick up TypeScript and linting from the top dir.

File: ui/package.json [hunk 1]
```diff
@@ -7,12 +7,12 @@
   "scripts": {
     "clean": "rm -rf node_modules yarn-error.log *.log build/ .bundle",
     "build": "npx vite build",
-    "format-js-eslint": "eslint 'src/**/*.{json,jsx,tsx,ts}' 'tests/**/*.ts' --fix",
```
Comment (huwshimi) [inline ui/package.json:10]: There are no files matching `tests/**/*.ts` so these calls were all failing.

File: ui/renovate.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
   "extends": [
-    "github>canonical-web-and-design/renovate-websites"
+    "github>canonical/renovate-apps"
```
Comment (huwshimi) [inline ui/renovate.json:3]: Use the new renovate config for apps.

File: ui/tsconfig.json [hunk 3]
```diff
@@ -1,14 +1,7 @@
...
     "outDir": "./public/js/dist",
-    "baseUrl": ".",
-    "paths": {
-      "api/*": ["src/api/*"],
-      "components/*": ["src/components/*"],
-      "pages/*": ["src/pages/*"],
-      "types/*": ["src/types/*"],
-      "util/*": ["src/util/*"]
-    },
+    "baseUrl": "src",
```
Comment (huwshimi) [inline ui/tsconfig.json:4]: This does the same thing as the manual paths above.

**Review by vladimir-cucu — APPROVED** [review]

**Review by BarcoMasile — APPROVED** [review]

**Pull request merged**